web_validation_dialog/static/src/js/web_validation_dialog.js

```diff
@@ -0,0 +1,112 @@
+                } else {
+                    return self.on_confirmed();
+                }
+            };
+            if (!this.node.attrs.special) {
+                return this.view.recursive_save().then(exec_action);
+            } else {
+                return exec_action();
+            }
+        },
+        open_pincode_dialog : function() {
+            var self = this;
+            new Dialog(self, {
+                title: _t('Validation'),
+                size : "small",
+                $content: QWeb.render('DialogValidation'),
+                buttons: [
+                            {
+                                text: _t("Ok"),
+                                classes: 'btn-primary',
+                                click: function() {
+                                   var curr_obj = this;
+                                   var password = this.$el.find("#pincode").val();
+                                   if (password) {
+                                       framework.blockUI();
+                                       var callback = self.validate_pincode(self.is_dialog_security, password);
+                                       callback.done(function(result) {
+                                           framework.unblockUI();
+                                           if (result) {
+                                               curr_obj.close();
+                                              self.on_confirmed();
+                                           } else {
+                                               Dialog.alert(self, _t("Invalid or Wrong Password! Contact your Administrator."));
+                                               return;
+                                           }
+                                       }).fail(function(error) {
+                                           framework.unblockUI();
+                                           Dialog.alert(self, _t("Either the password is wrong or the connection is lost! Contact your Administrator."));
+                                           return;
+                                       });
+                                   } else {
+                                       Dialog.alert(self, _t("Please Enter the Password."));
+                                       return;
+                                   }
+                                }
+                            },
+                            {
+                                text: _t('Cancel'),
+                                close: true
+                            }
+                        ]
+            }).open();
+        },
+        validate_pincode : function(field, value) {
+            var self = this;
+            var data_vals = {
+                "field": field,
+                "password": value,
+                "companyId": self.session.company_id
+            };
+            return new Model("res.company").call("check_security", [data_vals]);
```

Comment:
This obviously has to change once you change the backend method signature.

Reply:
nothing to do here?

Reply:
> nothing to do here?

@simahawk This method Call in python method [here](https://github.com/OCA/web/pull/728/files/ef07f9af45331733aba9bfb517991c0da7591a5e#diff-6113c0a76b7ebbb8aca53ab6c3696d75R13)

It's Check Only Password and return here.

Reply:
I think it would be more appropriate to actually add `.done()`, `.fail()` handlers here instead of defining them every time after the usage of this function.